src/runtime_src/hip/core/event.h

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Device, Inc. All rights reserved.
+#ifndef xrthip_event_h
+#define xrthip_event_h
+
+#include "stream.h"
+#include "common.h"
+#include "xrt/xrt_kernel.h"
+
+#include <condition_variable>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+namespace xrt::core::hip {
+class command
+{
+public:
+  enum class state : uint8_t
+  {
+    init,
+    recorded,
+    running,
+    completed,
+    error,
+    abort
+  };
+
+  enum class type : uint8_t
+  {
+    event,
+    buffer_copy,
+    kernel_start
+  };
+
+private:
+  std::shared_ptr<stream> cstream;
+  uint64_t ctime;
+  type ctype;
+  state cstate;
+
+public:
+  virtual bool submit() = 0;
+  virtual bool wait() = 0;
+};
+
+class event : public command
+{
+private:
+  std::vector<command*> dep_commands;
```

Comment:
do we need to keep shared_ptrs of commands?

Reply:
Can you please think of better name for this variable. (I know, i used same name in confluence :-) )

Reply:
Hi Vamshi, I changed it to "Sync_dependent_commands" please let me know if you think it works.